qiskit/_result.py

```diff
@@ -414,3 +414,22 @@ def get_qubitpol_vs_xval(self, nqubits, xvals_dict=None):
                     circuit_name, z_dicts[qubit_ind])
 
         return qubitpol, xvals
+
+
+def copy_qasm_from_qobj_into_result(qobj, result):
+    """Find the several QASM belonging to the Qobj experiments and copy them
+    into the corresponding result entries."""
+    for experiment in qobj.experiments:
+        name = experiment.header.name
+        qasm = getattr(experiment.header, 'compiled_circuit_qasm', None)
+        if qasm:
+            experiment_result = _find_experiment_result(result, name)
+            experiment_result['compiled_circuit_qasm'] = qasm
+
+
+def _find_experiment_result(result, name):
```

Comment:
Can you reeevaluate if this function is really needed? Seems we could avoid an extra function if it is made a one-liner with `next(x for x in result['result'] if blah)` (not really blocking anyway).

Reply:
I try to avoid this kind of one-liners since they are, usually, harder to read. I can move it inside `copy_qasm_from_qobj_into_result` as an auxiliary function or a lambda variable, though.

Reply:
I also dislike lots of small functions, but your choice.